source/gboardforensics/analysis/file.d

```diff
@@ -0,0 +1,127 @@
+module gboardforensics.analysis.file;
+
+import gboardforensics.gatherers;
+import gboardforensics.gatherers.dictionary;
+import gboardforensics.analysis;
+
+import std.stdio;
+import std.path : baseName;
+
+import d2sqlite3;
+
+/**
+ * Represents a GBoard file detector. This struct detects and gets the right
+ * file gatherer according to its filename and content.
+ */
+struct FileDetector
+{
+
+	/// SQLite3 database file signature
+	private static immutable SQLITE3_SIGNATURE = [
+		0x53, 0x51, 0x4C, 0x69, 0x74,
+		0x65, 0x20, 0x66, 0x6F, 0x72,
+		0x6D, 0x61, 0x74, 0x20, 0x33,
+		0x00
+	];
+
+	/**
+	 * Constructs a file detector using a file path
+	 *
+	 * Params:
+	 *   path = path to the file to be detected
+	 */
+	public this(string path)
+		in(exists(path) && isFile(path))
```

Comment:
As per the Dlang style:
> Put a space after in/out similar to if constraints

Also, you can have multiple `in` contracts. Totally preferential:
```d
in (path.exists())
in (path.isFile())
```